data/src/main/java/com/linkedin/data/schema/validation/ValidateDataAgainstSchema.java

```diff
@@ -173,9 +166,22 @@ protected void validateIterative(DataElement element)
       _fixed = element.getValue();
       UnrecognizedFieldMode unrecognizedFieldMode = _options.getUnrecognizedFieldMode();
       ObjectIterator it = new ObjectIterator(element, IterationOrder.POST_ORDER);
-      DataElement nextElement;
-      while ((nextElement = it.next()) != null)
+      DataElement nextElement = null;
+      while (true)
       {
+        try
+        {
+          if ((nextElement = it.next()) == null)
+          {
+            break;
+          }
+        }
+        catch (IllegalArgumentException e)
+        {
+          addMessage(nextElement, e.getMessage());
```

Comment:
Also log the exception for debugging.

Reply:
There has never been any logs in any Validators, either schema or data validators.

Also I am not sure if need them as theyare caught in the validation result